### DISCUSSION TITLE
If LDAP Is enabled, add a sync button beside users field when adding new assets

### DISCUSSION THREAD

**Discussion by MrCaspan** [snipe/snipe-it#9281] [Ideas]
Would be nice to have a little LDAP Sync button beside the New user button on the new assets page the fires off the LDAP sync feature

![2021 03 11 - 000195](https://user-images.githubusercontent.com/19710296/110836391-fa75c380-826d-11eb-9c6d-0fd4d70df6f2.png)

Then just change the button for a green check mark for 3 seconds to show it succeeded

Thoughts? Ideas?

**Comment by snipe** [comment]
What is the use case here?

**Comment by MrCaspan** [comment]
use case would be when I'm adding a new asset and assigning it to a new user. normally when I find a user is not in the system is when I'm adding a new asset and trying to assign it to someone. If they're not in the user list then I have to open up a new tab go to the user section Go to the LDAP section then synchronize it close the tab and go back to adding of the new asset and search for the user again. It would be nice if all the places where you can assign an asset to a user that there is a little synchronized icon there so you can synchronize LDAP without leaving the page

**Reply by snipe** [reply]
Ahhh, I see. The only place that could get weird is for very large directories that take some time to sync. (10k+ users in the directory, which isn't that uncommon for schools.)

It's a valid use-case, I'm just not sure this is exactly the best solution. Let me chew on this a bit.

**Reply by MrCaspan** [reply]
Yep NP just ideas.. I love throwing stuff at the walls!

**Reply by csayre** [reply]
Out of curiosity, why wouldn't a scheduled ldap sync work for this? You could set that as frequently as a minute if you wanted to.

**Reply by MrCaspan** [reply]
It is a possibility to do that. We pay for CPU cycles in the cloud so this would just cost us money for no real gain. Its just a nice to have not a must. In a basic work flow this is the point at which most users I would think would notice a user was missing so would be a great place to put the sync button.

I was raised in a house with a Mom that did ISO 9001 stuff and worked for Japanese companies with 5S philosophy and kaizen approach. For any worker they should have their work close at their reach. If you have to go somewhere else to do your job then A) if its a one off then not to worry B) if its part of your normal routine then fix it so its not out of the way. I can't help but see stuff this way. blessing and a curse!
